File: Components/Strata.Client.Core/main/typescript/Main/AbstractApplication.ts
import {IApplication} from "./IApplication";
import {IModelStore} from "../Presenter/IModelStore";
import {IPresenter} from "../Presenter/IPresenter";

export
abstract class AbstractApplication<M,V,P extends IPresenter<M,V>>
    implements IApplication
{
    private modelstore: IModelStore;
    private presenter: P;

    protected constructor()
    {
        this.modelstore = null;
        this.presenter = null;
    }

    protected initialize(modelstore: IModelStore,presenter: P)
    {
        this.modelstore = modelstore;
        this.presenter = presenter;

        this.modelstore.attach(this.presenter);
    }

    abstract start(): void;

    abstract stop(): void;

    protected getModelStore(): IModelStore
    {
        return this.modelstore;
    }

    protected getPresenter(): P
    {
        return this.presenter;
    }
}

File: Components/Strata.Client.Core/src/main/java/strata/client/core/main/AbstractApplication.java
//////////////////////////////////////////////////////////////////////////////
// AbstractApplication.java
//////////////////////////////////////////////////////////////////////////////

package strata.client.core.main;

import strata.client.core.presenter.IModelStore;
import strata.client.core.presenter.IPresenter;

public abstract
class AbstractApplication<M,V,P>
    implements IApplication
{
    private final IModelStore itsModelStore;
    private IPresenter<M,V> itsPresenter;
    private V                 itsView;

    protected
    AbstractApplication(IModelStore modelStore,IPresenter<M,V> presenter,V view)
    {
        itsModelStore = modelStore;
        itsPresenter = presenter;
        itsView = view;
    }

}

//////////////////////////////////////////////////////////////////////////////
